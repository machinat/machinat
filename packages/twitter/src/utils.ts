import { annotateNativeComponent } from '@machinat/core/renderer';
import TwitterUser from './User';
import TwitterChat from './Chat';
import TweetTarget from './TweetTarget';
import { TWITTER } from './constant';
import type {
  RawEntities,
  UrlEntity,
  MentionEntity,
  RawMedia,
  Media,
  MediaSource,
  DirectMessageSegmentValue,
  TweetSegmentValue,
} from './types';

export const makeTwitterComponent = annotateNativeComponent(TWITTER);

export const polishMentionEntity = (
  raw: RawEntities['user_mentions'][number]
): MentionEntity => ({
  user: new TwitterUser(raw.id_str),
  id: raw.id_str,
  name: raw.name,
  screenName: raw.screen_name,
  indices: raw.indices,
});

export const polishUrlEntity = (
  raw: RawEntities['urls'][number]
): UrlEntity => ({
  url: raw.url,
  displayUrl: raw.display_url,
  expandedUrl: raw.expanded_url,
  indices: raw.indices,
});

export const polishMedia = (raw: RawMedia): Media => {
  const videoInfo = raw.video_info;
  return {
    id: raw.id_str,
    type: raw.type,
    inlineUrl: raw.url,
    displayUrl: raw.display_url,
    expandedUrl: raw.expanded_url,
    indices: raw.indices,
    downloadUrl: raw.media_url_https,
    sizes: raw.sizes,
    sourceStatusId: raw.source_status_id_str,
    videoInfo: videoInfo
      ? {
          aspectRatio: videoInfo.aspect_ratio,
          durationMs: videoInfo.duration_millis,
          variants: videoInfo.variants.map((rawVariant) => ({
            url: rawVariant.url,
            bitrate: rawVariant.bitrate,
            contentType: rawVariant.content_type,
          })),
        }
      : null,
    additionalMediaInfo: raw.additional_media_info || null,
  } as Media;
};

export const annotateDmMedia = (media: MediaSource): MediaSource =>
  media.sourceType === 'id'
    ? media
    : {
        ...media,
        parameters: {
          ...media.parameters,
          media_category:
            media.parameters.media_category ||
            (media.type === 'photo'
              ? 'dm_image'
              : media.type === 'video'
              ? 'dm_video'
              : media.type === 'animated_gif'
              ? 'dm_gif'
              : undefined),
        },
      };

export const annotateTweetMedia = (media: MediaSource): MediaSource =>
  media.sourceType === 'id'
    ? media
    : {
        ...media,
        parameters: {
          ...media.parameters,
          media_category:
            media.parameters.media_category ||
            (media.type === 'photo'
              ? 'tweet_image'
              : media.type === 'video'
              ? 'tweet_video'
              : media.type === 'animated_gif'
              ? 'tweet_gif'
              : undefined),
        },
      };

export const createDmSegmentValue = (
  text?: string,
  media?: MediaSource
): DirectMessageSegmentValue => ({
  type: 'dm',
  request: {
    method: 'POST',
    href: '1.1/direct_messages/events/new.json',
    parameters: {
      event: {
        type: 'message_create',
        message_create: {
          target: { recipient_id: '' },
          message_data: {
            text,
            attachment: media
              ? {
                  type: 'media',
                  media: { id: '' },
                }
              : undefined,
          },
        },
      },
    },
  },
  accomplishRequest: (target: TwitterChat, request, mediaResults) => {
    // eslint-disable-next-line no-param-reassign
    request.parameters.event.message_create.target.recipient_id = target.id;

    if (mediaResults) {
      // eslint-disable-next-line no-param-reassign, prefer-destructuring
      request.parameters.event.message_create.message_data.attachment.media.id =
        mediaResults[0];
    }
    return request;
  },
  mediaSources: media ? [annotateDmMedia(media)] : null,
});

export const createTweetSegmentValue = (
  parameters: Record<string, any>,
  mediaSources?: null | MediaSource[],
  dmLink?:
    | boolean
    | string
    | { recipientId?: string; text?: string; welcomeMessageId?: string }
): TweetSegmentValue => ({
  type: 'tweet',
  request: {
    method: 'POST',
    href: '2/tweets',
    parameters,
  },
  mediaSources: mediaSources ? mediaSources.map(annotateTweetMedia) : null,
  accomplishRequest: (target: TweetTarget, request, mediaIds) => {
    const params = request.parameters;
    return {
      ...request,
      parameters: {
        ...request.parameters,
        reply: target.tweetId
          ? {
              in_reply_to_tweet_id: target.tweetId,
              exclude_reply_user_ids: params.reply?.exclude_reply_user_ids,
            }
          : undefined,
        media: mediaIds
          ? {
              media_ids: mediaIds,
              tagged_user_ids: params.media?.tagged_user_ids,
            }
          : undefined,
        direct_message_deep_link:
          typeof dmLink === 'string'
            ? dmLink
            : typeof dmLink === 'object'
            ? `https://twitter.com/messages/compose?recipient_id=${
                dmLink.recipientId || target.agentId
              }${
                dmLink.welcomeMessageId
                  ? `&welcome_message_id=${dmLink.welcomeMessageId}`
                  : ''
              }${dmLink.text ? `&text=${encodeURIComponent(dmLink.text)}` : ''}`
            : dmLink
            ? `https://twitter.com/messages/compose?recipient_id=${target.agentId}`
            : undefined,
      },
    };
  },
});

let lastTime = 0;
let count = 0;
export const getTimeId = (): string => {
  const now = Date.now();
  if (lastTime === now) {
    count += 1;
  } else {
    lastTime = now;
    count = 0;
  }
  return `${now}-${count}`;
};
