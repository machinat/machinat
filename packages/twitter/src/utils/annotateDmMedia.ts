import type { MediaSource, MediaType } from '../types.js';

const annotateDmMedia = (
  mediaType: MediaType,
  source: MediaSource,
): MediaSource =>
  source.type === 'id'
    ? source
    : {
        ...source,
        params: {
          ...source.params,
          media_category:
            source.params.media_category ||
            (mediaType === 'photo'
              ? 'dm_image'
              : mediaType === 'video'
              ? 'dm_video'
              : mediaType === 'animated_gif'
              ? 'dm_gif'
              : undefined),
        },
      };

export default annotateDmMedia;
