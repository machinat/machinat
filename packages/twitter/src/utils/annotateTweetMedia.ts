import type { MediaSource, MediaType } from '../types.js';

const annotateTweetMedia = (
  mediaType: MediaType,
  source: MediaSource,
): MediaSource =>
  source.type === 'id'
    ? source
    : {
        ...source,
        params: {
          ...source.params,
          media_category:
            source.params.media_category ||
            (mediaType === 'photo'
              ? 'tweet_image'
              : mediaType === 'video'
              ? 'tweet_video'
              : mediaType === 'animated_gif'
              ? 'tweet_gif'
              : undefined),
        },
      };

export default annotateTweetMedia;
