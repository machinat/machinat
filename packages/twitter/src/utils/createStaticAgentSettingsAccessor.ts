import { TwitterAgentSettings } from '../types';
import { AgentSettingsAccessorI } from '../interface';

const createStaticAgentSettingsAccessor = (
  settings: TwitterAgentSettings[]
): AgentSettingsAccessorI => ({
  getAgentSettings: async (agent) =>
    settings.find(({ userId }) => userId === agent.id) || null,
  getAgentSettingsBatch: async (agents) =>
    agents.map(
      (agent) => settings.find(({ userId }) => userId === agent.id) || null
    ),
});

export default createStaticAgentSettingsAccessor;
