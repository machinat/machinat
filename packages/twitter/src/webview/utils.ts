import type { ContextDetails } from '@sociably/auth';
import TwitterChat from '../Chat';
import TwitterUser from '../User';
import UserProfile from '../UserProfile';
import type { TwitterAuthContext, TwitterAuthData } from './types';

// eslint-disable-next-line import/prefer-default-export
export const getAuthContextDetails = ({
  agent: agentId,
  user: { id: userId, data: rawUserData },
}: TwitterAuthData): ContextDetails<TwitterAuthContext> => {
  return {
    agentId,
    channel: new TwitterUser(agentId),
    user: new TwitterUser(userId, rawUserData),
    thread: new TwitterChat(agentId, userId),
    userProfile: new UserProfile(rawUserData),
  };
};
