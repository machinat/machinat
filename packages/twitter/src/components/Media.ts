import { NativeElement } from '@machinat/core';
import { makeUnitSegment, UnitSegment } from '@machinat/core/renderer';
import { makeTwitterComponent } from '../utils';
import { TwitterSegmentValue, TwitterComponent, MediaType } from '../types';

/**
 * @category Props
 */
export type MediaProps = {
  /** Set to true if media asset will be reused for multiple Direct Messages. Default is false */
  shared?: boolean;
  /** The tag to store media asset with */
  assetTag?: string;
  /** A string enum value which identifies a media usecase. This identifier is used to enforce usecase specific constraints (e.g. file size, video duration) and enable advanced features */
  mediaCategory?: string;
  /** A comma-separated list of user IDs to set as additional owners allowed to use the returned media_id in Tweets or Cards. Up to 100 additional owners may be specified */
  additionalOwners?: string[];
  /** The media file url */
  url?: string;
  /** The uploaded media id */
  mediaId?: string;
  /** The file content to be upload */
  file?: Buffer | NodeJS.ReadableStream;
  /** The content type of the file. Required when uploading using `file` */
  fileType?: string;
  /** The file size in bytes. Required when uploading using `file` */
  fileSize?: number;
};

const makeMediaComponent = (name: string, type: MediaType) =>
  makeTwitterComponent(
    {
      [name](
        node: NativeElement<MediaProps, TwitterComponent<MediaProps>>,
        path: string
      ): UnitSegment<TwitterSegmentValue>[] {
        const {
          shared,
          assetTag,
          url,
          mediaId,
          file,
          fileSize,
          fileType,
          mediaCategory,
          additionalOwners,
        } = node.props;

        if (
          !(url || mediaId || file) ||
          (url && mediaId) ||
          (url && file) ||
          (file && mediaId)
        ) {
          throw new TypeError(
            'there should be exactly one of "url", "mediaId" or "file" prop'
          );
        }
        if (file && (!fileType || !fileSize)) {
          throw new TypeError(
            '"fileType" and "fileSize" props are required when using "file" source'
          );
        }

        const parameters = {
          shared: shared ? 'true' : undefined,
          media_category: mediaCategory,
          additional_owners: additionalOwners
            ? additionalOwners.join(',')
            : undefined,
        };

        return [
          makeUnitSegment(node, path, {
            type: 'media',
            media: mediaId
              ? {
                  sourceType: 'id',
                  type,
                  id: mediaId,
                  parameters,
                }
              : url
              ? {
                  sourceType: 'url',
                  type,
                  url,
                  parameters,
                }
              : file
              ? {
                  type,
                  sourceType: 'file',
                  parameters: {
                    ...parameters,
                    total_bytes: String(fileSize),
                    media_type: fileType,
                  },
                  fileData: file,
                  fileInfo: {
                    contentType: fileType,
                    knownLength: fileSize,
                  },
                  assetTag,
                }
              : (null as never),
          }),
        ];
      },
    }[name]
  );

/**
 * Send a photo by direct message or attach it to a tweet
 * @category Component
 * @props {@link MediaProps}
 * @guides Check official [guide](https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/overview).
 */
export const Photo: TwitterComponent<
  MediaProps,
  UnitSegment<TwitterSegmentValue>
> = makeMediaComponent('Photo', 'photo');

/**
 * Send a video by direct message or attach it to a tweet
 * @category Component
 * @props {@link MediaProps}
 * @guides Check official [guide](https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/overview).
 */
export const Video: TwitterComponent<
  MediaProps,
  UnitSegment<TwitterSegmentValue>
> = makeMediaComponent('Video', 'video');

/**
 * Send an animated gif by direct message or attach it to a tweet
 * @category Component
 * @props {@link MediaProps}
 * @guides Check official [guide](https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/overview).
 */
export const AnimatedGif: TwitterComponent<
  MediaProps,
  UnitSegment<TwitterSegmentValue>
> = makeMediaComponent('AnimatedGif', 'animated_gif');
