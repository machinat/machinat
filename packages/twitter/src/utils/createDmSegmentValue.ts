import TwitterChat from '../Chat.js';
import type { MediaAttachment, DirectMessageSegmentValue } from '../types.js';
import annotateDmMedia from './annotateDmMedia.js';

const createDmSegmentValue = (
  text?: string,
  media?: MediaAttachment,
): DirectMessageSegmentValue => ({
  type: 'dm',
  request: {
    method: 'POST',
    url: '1.1/direct_messages/events/new.json',
    params: {
      event: {
        type: 'message_create',
        message_create: {
          target: { recipient_id: '' },
          message_data: {
            text,
            attachment: media
              ? {
                  type: 'media',
                  media: { id: '' },
                }
              : undefined,
          },
        },
      },
    },
  },
  accomplishRequest: (target: TwitterChat, request, mediaResults) => {
    // eslint-disable-next-line no-param-reassign
    request.params.event.message_create.target.recipient_id = target.userId;

    if (mediaResults) {
      // eslint-disable-next-line no-param-reassign
      request.params.event.message_create.message_data.attachment.media.id =
        mediaResults[0];
    }
    return request;
  },
  mediaSources: media ? [annotateDmMedia(media.type, media.source)] : null,
});

export default createDmSegmentValue;
