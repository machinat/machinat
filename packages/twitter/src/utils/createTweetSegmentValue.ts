import TweetTarget from '../TweetTarget';
import type { MediaAttachment, TweetSegmentValue } from '../types';
import annotateTweetMedia from './annotateTweetMedia';

const createTweetSegmentValue = (
  parameters: Record<string, any>,
  media?: null | MediaAttachment[],
  dmLink?:
    | boolean
    | string
    | { recipientId?: string; text?: string; welcomeMessageId?: string }
): TweetSegmentValue => ({
  type: 'tweet',
  request: {
    method: 'POST',
    href: '2/tweets',
    parameters,
  },
  mediaSources: media
    ? media.map(({ source, type }) => annotateTweetMedia(type, source))
    : null,
  accomplishRequest: (target: TweetTarget, request, mediaIds) => {
    const params = request.parameters;
    return {
      ...request,
      parameters: {
        ...request.parameters,
        reply: target.tweetId
          ? {
              in_reply_to_tweet_id: target.tweetId,
              exclude_reply_user_ids: params.reply?.exclude_reply_user_ids,
            }
          : undefined,
        media: mediaIds
          ? {
              media_ids: mediaIds,
              tagged_user_ids: params.media?.tagged_user_ids,
            }
          : undefined,
        direct_message_deep_link:
          typeof dmLink === 'string'
            ? dmLink
            : typeof dmLink === 'object'
            ? `https://twitter.com/messages/compose?recipient_id=${
                dmLink.recipientId || target.agentId
              }${
                dmLink.welcomeMessageId
                  ? `&welcome_message_id=${dmLink.welcomeMessageId}`
                  : ''
              }${dmLink.text ? `&text=${encodeURIComponent(dmLink.text)}` : ''}`
            : dmLink
            ? `https://twitter.com/messages/compose?recipient_id=${target.agentId}`
            : undefined,
      },
    };
  },
});

export default createTweetSegmentValue;
