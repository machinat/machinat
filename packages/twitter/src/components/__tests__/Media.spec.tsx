import Sociably from '@sociably/core';
import { isNativeType } from '@sociably/core/utils';
import { Photo, Video, AnimatedGif } from '../Media.js';
import { renderUnitElement } from './utils.js';

const mediaNameComponentPairs = [Photo, Video, AnimatedGif].map(
  (Media): [string, typeof Photo] => [Media.$$name, Media],
);

test.each(mediaNameComponentPairs)('is a valid Component', (_, Media) => {
  expect(isNativeType(<Media />)).toBe(true);
  expect(Media.$$platform).toBe('twitter');
});

test.each(mediaNameComponentPairs)('%s rendering', async (mediaName, Media) => {
  await expect(
    renderUnitElement(<Media mediaId="12345" />),
  ).resolves.toMatchSnapshot();
  await expect(
    renderUnitElement(<Media url="http://foo.io/bar.mp4" shared />),
  ).resolves.toMatchSnapshot();
  await expect(
    renderUnitElement(
      <Media
        file={{
          data: Buffer.from('foo'),
          contentType:
            Media === Photo
              ? 'image/jpeg'
              : Media === Video
              ? 'video/mp4'
              : 'image/gif',
          contentLength: 1234,
        }}
        assetTag={`My${mediaName}`}
        shared
        additionalOwners={['12345']}
        mediaCategory={
          Media === Photo ? 'dm_image' : Media === Video ? 'dm_video' : 'dm_gif'
        }
      />,
    ),
  ).resolves.toMatchSnapshot();
});

test.each(mediaNameComponentPairs)(
  'throw if no media source is provided',
  async (_, Media) => {
    await expect(renderUnitElement(<Media />)).rejects.toThrow(
      'there should be exactly one of "url", "mediaId" or "file" prop',
    );
  },
);

test.each(mediaNameComponentPairs)(
  'throw if multiple media source are provided',
  async (_, Media) => {
    const fileSource = {
      data: Buffer.from('foo'),
      contentType: 'image/jpeg',
      contentLength: 1234,
    };
    await expect(
      renderUnitElement(<Media mediaId="12345" url="http://..." />),
    ).rejects.toThrow(
      'there should be exactly one of "url", "mediaId" or "file" prop',
    );
    await expect(
      renderUnitElement(<Media mediaId="12345" file={fileSource} />),
    ).rejects.toThrow(
      'there should be exactly one of "url", "mediaId" or "file" prop',
    );
    await expect(
      renderUnitElement(<Media url="http://..." file={fileSource} />),
    ).rejects.toThrow(
      'there should be exactly one of "url", "mediaId" or "file" prop',
    );
    await expect(
      renderUnitElement(
        <Media mediaId="12345" url="http://..." file={fileSource} />,
      ),
    ).rejects.toThrow(
      'there should be exactly one of "url", "mediaId" or "file" prop',
    );
  },
);
