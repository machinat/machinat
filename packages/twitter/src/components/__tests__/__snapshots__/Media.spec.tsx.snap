// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedGif rendering 1`] = `
Array [
  Object {
    "node": <AnimatedGif
      mediaId="12345"
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "id": "12345",
        "parameters": Object {
          "additional_owners": undefined,
          "media_category": undefined,
          "shared": undefined,
        },
        "sourceType": "id",
        "type": "animated_gif",
      },
      "type": "media",
    },
  },
]
`;

exports[`AnimatedGif rendering 2`] = `
Array [
  Object {
    "node": <AnimatedGif
      shared={true}
      url="http://foo.io/bar.mp4"
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "parameters": Object {
          "additional_owners": undefined,
          "media_category": undefined,
          "shared": "true",
        },
        "sourceType": "url",
        "type": "animated_gif",
        "url": "http://foo.io/bar.mp4",
      },
      "type": "media",
    },
  },
]
`;

exports[`AnimatedGif rendering 3`] = `
Array [
  Object {
    "node": <AnimatedGif
      additionalOwners={
        Array [
          "12345",
        ]
      }
      assetTag="MyAnimatedGif"
      file={
        Object {
          "data": Array [
            102,
            111,
            111,
          ],
          "type": "Buffer",
        }
      }
      fileSize={1234}
      fileType="image/gif"
      mediaCategory="dm_gif"
      shared={true}
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "assetTag": "MyAnimatedGif",
        "fileData": Object {
          "data": Array [
            102,
            111,
            111,
          ],
          "type": "Buffer",
        },
        "fileInfo": Object {
          "contentType": "image/gif",
          "knownLength": 1234,
        },
        "parameters": Object {
          "additional_owners": "12345",
          "media_category": "dm_gif",
          "media_type": "image/gif",
          "shared": "true",
          "total_bytes": "1234",
        },
        "sourceType": "file",
        "type": "animated_gif",
      },
      "type": "media",
    },
  },
]
`;

exports[`Photo rendering 1`] = `
Array [
  Object {
    "node": <Photo
      mediaId="12345"
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "id": "12345",
        "parameters": Object {
          "additional_owners": undefined,
          "media_category": undefined,
          "shared": undefined,
        },
        "sourceType": "id",
        "type": "photo",
      },
      "type": "media",
    },
  },
]
`;

exports[`Photo rendering 2`] = `
Array [
  Object {
    "node": <Photo
      shared={true}
      url="http://foo.io/bar.mp4"
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "parameters": Object {
          "additional_owners": undefined,
          "media_category": undefined,
          "shared": "true",
        },
        "sourceType": "url",
        "type": "photo",
        "url": "http://foo.io/bar.mp4",
      },
      "type": "media",
    },
  },
]
`;

exports[`Photo rendering 3`] = `
Array [
  Object {
    "node": <Photo
      additionalOwners={
        Array [
          "12345",
        ]
      }
      assetTag="MyPhoto"
      file={
        Object {
          "data": Array [
            102,
            111,
            111,
          ],
          "type": "Buffer",
        }
      }
      fileSize={1234}
      fileType="image/jpeg"
      mediaCategory="dm_image"
      shared={true}
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "assetTag": "MyPhoto",
        "fileData": Object {
          "data": Array [
            102,
            111,
            111,
          ],
          "type": "Buffer",
        },
        "fileInfo": Object {
          "contentType": "image/jpeg",
          "knownLength": 1234,
        },
        "parameters": Object {
          "additional_owners": "12345",
          "media_category": "dm_image",
          "media_type": "image/jpeg",
          "shared": "true",
          "total_bytes": "1234",
        },
        "sourceType": "file",
        "type": "photo",
      },
      "type": "media",
    },
  },
]
`;

exports[`Video rendering 1`] = `
Array [
  Object {
    "node": <Video
      mediaId="12345"
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "id": "12345",
        "parameters": Object {
          "additional_owners": undefined,
          "media_category": undefined,
          "shared": undefined,
        },
        "sourceType": "id",
        "type": "video",
      },
      "type": "media",
    },
  },
]
`;

exports[`Video rendering 2`] = `
Array [
  Object {
    "node": <Video
      shared={true}
      url="http://foo.io/bar.mp4"
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "parameters": Object {
          "additional_owners": undefined,
          "media_category": undefined,
          "shared": "true",
        },
        "sourceType": "url",
        "type": "video",
        "url": "http://foo.io/bar.mp4",
      },
      "type": "media",
    },
  },
]
`;

exports[`Video rendering 3`] = `
Array [
  Object {
    "node": <Video
      additionalOwners={
        Array [
          "12345",
        ]
      }
      assetTag="MyVideo"
      file={
        Object {
          "data": Array [
            102,
            111,
            111,
          ],
          "type": "Buffer",
        }
      }
      fileSize={1234}
      fileType="video/mp4"
      mediaCategory="dm_video"
      shared={true}
    />,
    "path": "$",
    "type": "unit",
    "value": Object {
      "media": Object {
        "assetTag": "MyVideo",
        "fileData": Object {
          "data": Array [
            102,
            111,
            111,
          ],
          "type": "Buffer",
        },
        "fileInfo": Object {
          "contentType": "video/mp4",
          "knownLength": 1234,
        },
        "parameters": Object {
          "additional_owners": "12345",
          "media_category": "dm_video",
          "media_type": "video/mp4",
          "shared": "true",
          "total_bytes": "1234",
        },
        "sourceType": "file",
        "type": "video",
      },
      "type": "media",
    },
  },
]
`;
